src/rtctools/optimization/seed_mixin.py

```diff
@@ -0,0 +1,129 @@
+from pathlib import Path
+
+from rtctools.data.timeseries import (
+    check_times_are_equidistant,
+    check_times_are_increasing,
+    fill_nan_in_timeseries,
+    get_timeseries_from_csv,
+    get_timeseries_from_pi,
+)
+from rtctools.optimization.csv_mixin import CSVMixin
+from rtctools.optimization.goal_programming_mixin import GoalProgrammingMixin
+from rtctools.optimization.homotopy_mixin import HomotopyMixin
+from rtctools.optimization.optimization_problem import OptimizationProblem
+from rtctools.optimization.pi_mixin import PIMixin
+from rtctools.optimization.timeseries import Timeseries
+
+
+class SeedMixin(OptimizationProblem):
+    """
+    Adds options for reading a seed from a given file to your optimization problem.
+    """
+
+    def __init__(self, **kwargs):
+        self._seed_timeseries: Path = None
+        super().__init__(**kwargs)
+
+    def _use_seed_timeseries(self):
+        """Return True if a seed from a timeseries is used for the current run."""
+        if self._seed_timeseries is None:
+            return False
+        if isinstance(self, GoalProgrammingMixin):
+            if not self._gp_first_run:
+                return False
+        if isinstance(self, HomotopyMixin):
```

Comment:
Ideally there would be no horizontal dependencies between mixins.

Reply:
Yes true, although I could not find an easy solution for this. We use a similar approach in HomotopyMixin although it is not ideal there either. I'll try again if I can get rid of the dependencies.

Reply:
The user can now overwrite the use_seed_file method (by default returns True). This avoids mixin dependencies and only requires a few lines of code from the user.